### DISCUSSION TITLE
go.d SNMP sanity

### DISCUSSION THREAD

**Discussion by nikobearrr** [netdata/netdata#14534] [Q&A] [answered]
I added one of my hosts to be monitored by SNMP. However in the logs I get this:

```
2023-02-15 09:37:03: go.d INFO: snmp[host03] hostname=host03,port=161,snmp_version=2c,community=public
2023-02-15 09:37:03: go.d ERROR: snmp[host03] cannot get SNMP data: error verifying packet sanity: Got 484 Expected: 574
2023-02-15 09:37:03: go.d ERROR: snmp[host03] error verifying packet sanity: Got 484 Expected: 574
2023-02-15 09:37:03: go.d ERROR: snmp[host03] check failed
```

When I do a `snmpwalk` I get the SNMP data normally. I tried searching in this repository as well as the `go.d-plugin` repository for `sanity` but I cannot find anything related.
I tried restarting my device and also enabling/disabling SNMP. 
It only happened after running my device for a while (it was able to fetch the metrics before that). Now it has stopped and no restarts (both netdata and my host) makes it work.

Can you suggest an action I can do to verify my device is indeed sane and make the snmp collector happy again?

**Comment by ilyam8** [comment, marked as answer]
I see the error comes from the [unmarshalHeader()](https://github.com/gosnmp/gosnmp/blob/7f6d7056e90d3ecc246b5399ed0153f99344b5bb/marshal.go#L991-L1015) function. It happens because the actual length (Got) is not equal to the calculated (Expected, see [parseLength()](https://github.com/gosnmp/gosnmp/blob/7f6d7056e90d3ecc246b5399ed0153f99344b5bb/helper.go#L587-L599)).

I don't know how to solve the problem. I think you need to ask [gosnmp developers/community](https://github.com/gosnmp/gosnmp) (the package Netdata uses).
